pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -645,6 +644,10 @@ func (s *Service) getSecurityGroupIngressRules(role infrav1.SecurityGroupRole) (
 		return append(cniRules, rules...), nil
 
 	case infrav1.SecurityGroupNode:
+		cidrBlocks := []string{services.AnyIPv4CidrBlock}
```

Comment:
For the record: if someone uses an IPv6-enabled VPC, the result will be surprising (see some lines below where we allow all incoming traffic from IPv6 sources). Okay to leave that part out for now, since there's not much contribution interest in IPv6, nor is it used widely anywhere.

Side note (personal opinion): we should not distinguish IPv4/IPv6 with separate field names. Instead, allow parsing `NodePortIngressRuleCidrBlocks []string` as either. If it contains an IPv6 CIDR, deal with it.

Reply:
I agree about second point. But it looks like for me that `[]string` is not suite for this task. It will require parsing strings during provisioning. But not sure that usage of like `[]CidrBlock` where CidrBlock is interface will be better.
Thank you!